test/functional/core/startup_spec.lua

```diff
@@ -472,7 +473,7 @@ describe('user config init', function()
     end)
 
     it('loads custom lua config and does not set $MYVIMRC', function()
-      clear{ args={'-u', custom_lua_path }, env={ XDG_CONFIG_HOME=xconfig }}
+      clear{ args={'-u', custom_lua_path }, env={ XDG_CONFIG_HOME=xconfig ; XDG_DATA_HOME=xconfig; VIMRUNTIME=xconfig}}
```

Comment:
What's weird is XDG_DATA_HOME should be set on line 12 in cmake `set(ENV{XDG_DATA_HOME} ${BUILD_DIR}/Xtest_xdg/share)`

Reply:
I guess this points at `INFO  2021-02-18T11:07:10.488 1012134 emsg_multiline:681: E484: Can't open file Xhome/Xconfig/syntax/syntax.vim`

Reply:
Here's the full RTP path *without* manually overriding XDG_DATA_HOME:
```
INFO  2021-02-18T11:09:34.293 1014444 set_init_2:754: startup runtimepath/packpath value: Xhome/Xconfig/nvim,/etc/xdg/nvim,Xhome/Xconfig/nvim/site,/nix/store/mzbylj1zprn18m7h2nsf6vmfafkvz54r-cmake-3.19.3/share/nvim/site,/nix/store/yn0mz3d30ryx5ids1lf3c4jxgi7r2rci-gettext-0.21/share/nvim/site,/nix/store/6c2s51yjkb2nab010b8dhdldh28dhnvj-pkg-config-wrapper-0.29.2/share/nvim/site,/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.7/share/nvim/site,/nix/store/khx9absslbabgfwbzxig5258nh138k1p-include-what-you-use-0.14/share/nvim/site,/nix/store/whmd7hgfnfy27ab8d9m7z0d05v0h0yai-fish-3.1.2/share/nvim/site,/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/share/nvim/site,/nix/store/r19q2xb4qw4l9v13hi1zgjrw5sd5dg0x-gnused-4.8/share/nvim/site,/nix/store/f89071fz20w31wjf116h5l3m5j8qkwld-groff-1.22.4/share/nvim/site,/nix/store/8i2wxn1gbjvaq0l68yhz4dsfpdnkai59-man-db-2.9.3/share/nvim/site,/nix/store/c03izgzw69sz495r2ck3admhq01p0q05-nodejs-14.15.5/share/nvim/site,/nix/store/jm7kyn357z32hz5nr1pgkpgyfyiai4b0-python-2.7.18-env/share/nvim/site,/nix/store/mcxhdbhz0wqi3icx6lvc33iaggd02v6v-patchelf-0.12/share/nvim/site,Xhome/Xconfig,/home/michael/Repositories/neovim_development/neovim/build/lib/nvim,/nix/store/mcxhdbhz0wqi3icx6lvc33iaggd02v6v-patchelf-0.12/share/nvim/site/after,/nix/store/jm7kyn357z32hz5nr1pgkpgyfyiai4b0-python-2.7.18-env/share/nvim/site/after,/nix/store/c03izgzw69sz495r2ck3admhq01p0q05-nodejs-14.15.5/share/nvim/site/after,/nix/store/8i2wxn1gbjvaq0l68yhz4dsfpdnkai59-man-db-2.9.3/share/nvim/site/after,/nix/store/f89071fz20w31wjf116h5l3m5j8qkwld-groff-1.22.4/share/nvim/site/after,/nix/store/r19q2xb4qw4l9v13hi1zgjrw5sd5dg0x-gnused-4.8/share/nvim/site/after,/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/share/nvim/site/after,/nix/store/whmd7hgfnfy27ab8d9m7z0d05v0h0yai-fish-3.1.2/share/nvim/site/after,/nix/store/khx9absslbabgfwbzxig5258nh138k1p-include-what-you-use-0.14/share/nvim/site/after,/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.7/share/nvim/site/after,/nix/store/6c2s51yjkb2nab010b8dhdldh28dhnvj-pkg-config-wrapper-0.29.2/share/nvim/site/after,/nix/store/yn0mz3d30ryx5ids1lf3c4jxgi7r2rci-gettext-0.21/share/nvim/site/after,/nix/store/mzbylj1zprn18m7h2nsf6vmfafkvz54r-cmake-3.19.3/share/nvim/site/after,Xhome/Xconfig/nvim/site/after,/etc/xdg/nvim/after,Xhome/Xconfig/nvim/after
INFO  2021-02-18T11:09:34.307 1014444 emsg_multiline:681: E484: Can't open file Xhome/Xconfig/syntax/syntax.vim
```

Reply:
With: 
```
INFO  2021-02-18T11:11:48.061 1016497 set_init_2:754: startup runtimepath/packpath value: Xhome/Xconfig/nvim,/etc/xdg/nvim,Xhome/Xconfig/nvim/site,/nix/store/mzbylj1zprn18m7h2nsf6vm
fafkvz54r-cmake-3.19.3/share/nvim/site,/nix/store/yn0mz3d30ryx5ids1lf3c4jxgi7r2rci-gettext-0.21/share/nvim/site,/nix/store/6c2s51yjkb2nab010b8dhdldh28dhnvj-pkg-config-wrapper-0.29.2
/share/nvim/site,/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.7/share/nvim/site,/nix/store/khx9absslbabgfwbzxig5258nh138k1p-include-what-you-use-0.14/share/nvim/site,/nix
/store/whmd7hgfnfy27ab8d9m7z0d05v0h0yai-fish-3.1.2/share/nvim/site,/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/share/nvim/site,/nix/store/r19q2xb4qw4l9v13hi1zgjrw5sd5dg0
x-gnused-4.8/share/nvim/site,/nix/store/f89071fz20w31wjf116h5l3m5j8qkwld-groff-1.22.4/share/nvim/site,/nix/store/8i2wxn1gbjvaq0l68yhz4dsfpdnkai59-man-db-2.9.3/share/nvim/site,/nix/s
tore/c03izgzw69sz495r2ck3admhq01p0q05-nodejs-14.15.5/share/nvim/site,/nix/store/jm7kyn357z32hz5nr1pgkpgyfyiai4b0-python-2.7.18-env/share/nvim/site,/nix/store/mcxhdbhz0wqi3icx6lvc33i
aggd02v6v-patchelf-0.12/share/nvim/site,Xhome/Xconfig,/home/michael/Repositories/neovim_development/neovim/build/lib/nvim,/nix/store/mcxhdbhz0wqi3icx6lvc33iaggd02v6v-patchelf-0.12/s
hare/nvim/site/after,/nix/store/jm7kyn357z32hz5nr1pgkpgyfyiai4b0-python-2.7.18-env/share/nvim/site/after,/nix/store/c03izgzw69sz495r2ck3admhq01p0q05-nodejs-14.15.5/share/nvim/site/a
fter,/nix/store/8i2wxn1gbjvaq0l68yhz4dsfpdnkai59-man-db-2.9.3/share/nvim/site/after,/nix/store/f89071fz20w31wjf116h5l3m5j8qkwld-groff-1.22.4/share/nvim/site/after,/nix/store/r19q2xb
4qw4l9v13hi1zgjrw5sd5dg0x-gnused-4.8/share/nvim/site/after,/nix/store/fbkni3i1sjfxnfld8f6dmrri1f0q5y6i-gnugrep-3.6/share/nvim/site/after,/nix/store/whmd7hgfnfy27ab8d9m7z0d05v0h0yai-
fish-3.1.2/share/nvim/site/after,/nix/store/khx9absslbabgfwbzxig5258nh138k1p-include-what-you-use-0.14/share/nvim/site/after,/nix/store/wkw6fsjasr7jbbrlakxxpbiapa8hws42-python3-3.8.
7/share/nvim/site/after,/nix/store/6c2s51yjkb2nab010b8dhdldh28dhnvj-pkg-config-wrapper-0.29.2/share/nvim/site/after,/nix/store/yn0mz3d30ryx5ids1lf3c4jxgi7r2rci-gettext-0.21/share/nv
im/site/after,/nix/store/mzbylj1zprn18m7h2nsf6vmfafkvz54r-cmake-3.19.3/share/nvim/site/after,Xhome/Xconfig/nvim/site/after,/etc/xdg/nvim/after,Xhome/Xconfig/nvim/after
INFO  2021-02-18T11:11:48.071 1016497 emsg_multiline:681: E484: Can't open file Xhome/Xconfig/syntax/syntax.vim
```